File: qqbot-module-operations/src/main/java/com/mieai/qqbot/admin/events/InboxEventDetailResponse.java
package com.mieai.qqbot.admin.events;

import java.time.Instant;
import java.util.UUID;

/** Detailed Inbox event view. The raw payload is deliberately kept as text. */
public record InboxEventDetailResponse(
        UUID id,
        String environment,
        UUID botId,
        String botDisplayName,
        String appId,
        String eventType,
        String platformEventId,
        String status,
        int attempt,
        Instant availableAt,
        Instant receivedAt,
        Instant updatedAt,
        String lastError,
        String payload,
        boolean payloadTruncated) {
}

File: qqbot-module-operations/src/main/kotlin/com/mieai/qqbot/admin/events/InboxEventDetailResponse.kt
package com.mieai.qqbot.admin.events

import java.time.Instant
import java.util.UUID

/** Detailed Inbox event view. The raw payload is deliberately kept as text. */
data class InboxEventDetailResponse(
    val id: UUID,
    val environment: String,
    val botId: UUID,
    val botDisplayName: String?,
    val appId: String?,
    val eventType: String,
    val platformEventId: String,
    val status: String,
    val attempt: Int,
    val availableAt: Instant,
    val receivedAt: Instant,
    val updatedAt: Instant,
    val lastError: String?,
    val payload: String,
    val payloadTruncated: Boolean,
)
